recommender/engines/persistent_mixin.py

```diff
@@ -17,35 +15,49 @@ class Persistent(ABC):
     """
 
     @classmethod
-    def load(cls, version: str = None) -> Any:
+    def check_existence(cls, version: str = None) -> Any:
```

Comment:
rename to fetch_latest_component? you're not only checking, you're also loading the latest one

Reply:
Also, return type should be Optional[MLComponent]

Reply:
From a technical point of view - yes, but that change causes a lot of typing-related warnings in `rl_inference_component` and `ncf_inference_component`. Functions expect one of the `MLComponent` types and receive `MLComponent | None`. So it's not such a small change